### DISCUSSION TITLE
React-Aria-Components Resizable Table - forbid the horizontal scroll?

### DISCUSSION THREAD

**Discussion by paroxyzm** [adobe/react-spectrum#6457] [General]
Hello good people!

I'm using a `Table` component together with `ResizableTableContainer` to make columns resizable. All works great, but I have a use-case where we'd like to "freeze" the table-width to whatever is currently set by the `ResizableTableContainer`.  In other words prevent horizontal scrolls from appearing when expanding the columns AND prevent "empty" space when shrinking the columns too much.

I was wondering if you have any advice, or maybe have encountered this issue/use-case before?
 
<img width="536" alt="shrinking too much" src="https://github.com/adobe/react-spectrum/assets/6005411/4e8419bc-4624-4360-b25d-8369eae7dcec">
<img width="455" alt="scrollbar appears" src="https://github.com/adobe/react-spectrum/assets/6005411/843ec4a5-de82-42ea-87b5-6fa99f7007a1">

I'm thinking about controlling the `width` of the individual columns and getting their widths via `onResize` callback, but that seems to require a separate algorithm to the one already implemented. 

Thanks in advance!

**Comment by snowystinger** [comment]
Behavior currently matches Finder on Apple, including the expand and shrink past table bounds. You could theoretically implement this by measuring the elements yourself and controlling the width yourself as you noted.
I think that's the best thing to do right now.

I've thought about allowing injection of other column sizing algorithms, but haven't had time to get into pitfalls around it or what the API could look like. It's a deceptively complex issue. Especially when applying a controlled state.
